include/pybind11/pybind11.h

```diff
@@ -1768,23 +1785,64 @@ iterator make_key_iterator(Iterator first, Sentinel last, Extra &&... extra) {
             }, std::forward<Extra>(extra)..., Policy);
     }
 
-    return cast(state{first, last, true});
+    return state{first, last, true};
+}
+
+/// Makes a python iterator from a first and past-the-end C++ InputIterator.
+template <return_value_policy Policy = return_value_policy::reference_internal,
+    typename Iterator,
+    typename Sentinel,
+    typename ValueType = decltype(*std::declval<Iterator>()),
+    typename... Extra>
+[[deprecated("Superseded by make_iterator_ng")]]
```

Comment:
The `[[deprecated]]` attribute is C++14. Use `PYBIND11_DEPRECATED` macro instead.